utils/fetchProfile.ts

```diff
@@ -4,18 +4,15 @@ import { eip165ABI } from '@/shared/abis/eip165ABI'
 import { PROVIDERS } from '@/types/enums'
 import { getDataABI } from '@/shared/abis/getDataABI'
 
-export const fetchProfile = async () => {
-  // get profile address from router (url)
-  const router = useRouter()
-  const { setAddress, setProfile } = useProfileStore()
-  const profileAddress = router.currentRoute.value.params.profileAddress
+export const fetchProfile = async (profileAddress: Address) => {
+  const { contract, isEoA } = useWeb3(PROVIDERS.RPC)
 
-  // valid address check
-  assertAddress(profileAddress)
-  setAddress(profileAddress)
+  // EoA check
+  if (await isEoA(profileAddress)) {
+    throw new Error('The profile is an EoA')
+  }
```

Comment:
if user attempts to connect an EOA, very likely he/she is using Metamask or an other wallet. Maybe we can adjust the UI to tell the user about the extension and UPs etc.? [sharing thoughts here for later]

Reply:
Makes sense, atm we show generic error modal but I will show more personalized message so user is aware.